### PR TITLE
gql explorer type class->fc

### PR DESCRIPTION
**Pull request by jackkav**
Highlights

- no behaviours just gql type handling
- took a little logic to parse out all the typescript ignores
image

### CODE REVIEW

File: packages/insomnia/src/ui/components/graph-ql-explorer/graph-ql-explorer-type.tsx [hunk 1]
```diff
@@ -15,94 +13,64 @@ interface Props {
...
-        {this.renderFieldsMaybe()}
-      </div>
-    );
-  }
-}
+  return (
+    <div className="graphql-explorer__type">
+      <MarkdownPreview markdown={markdown} />
+      {hasSchemaAndTypes ?
+        <Fragment>
```
Comment (marckong) [inline packages/insomnia/src/ui/components/graph-ql-explorer/graph-ql-explorer-type.tsx:53]: non blocker - <></> can be used here

File: packages/insomnia/src/ui/components/graph-ql-explorer/graph-ql-explorer-type.tsx [hunk 2]
```diff
@@ -15,94 +13,64 @@ interface Props {
...
-
-    if (schema === null) {
-      return null;
-    }
-
-    let title = 'Types';
-    let types: readonly GraphQLInterfaceType[] | GraphQLObjectType[] = [];
-
+export const GraphQLExplorerType: FC<Props> = ({ schema, type, onNavigateType, onNavigateField }) => {
+  const getTitle = () => {
```
Comment (marckong) [inline packages/insomnia/src/ui/components/graph-ql-explorer/graph-ql-explorer-type.tsx:17]: nit pick - if this function doesn't need to mutate the state but just return a value based on props, I wonder why these need to be declared as functions inside of the components (perhaps will change the ref every rerendering). 
```ts
 function getTitle(gqlType: GraphQLType): string {
     if (gqlType instanceof GraphQLUnionType) {
      return 'Possible Types';
    }
    if (gqlType instanceof GraphQLInterfaceType) {
      return 'Implementations';
    }
    if (gqlType instanceof GraphQLObjectType) {
      return 'Implements';
    }
    return 'Types';
 }

export const GraphQLExplorerType: FC<Props> = ({ schema, type, onNavigateType, onNavigateField }) => {
  // ...
  const title = getTitle(type);
  return (
    <div className="graphql-explorer__type">the rest...</div>
  );
};
```

or you can arguably just write a function hook and register this locally declared variable (which is based off prop) as a React state as well.

**Review by marckong — COMMENTED** [review]
It's great initiative! Since we aren't adding unit tests, I will run a smoke test on my end.

File: packages/insomnia/src/ui/components/graph-ql-explorer/graph-ql-explorer-type.tsx [hunk 3]
```diff
@@ -15,94 +13,64 @@ interface Props {
...
+    }
+    if (type instanceof GraphQLInterfaceType) {
+      return 'Implementations';
+    }
+    if (type instanceof GraphQLObjectType) {
+      return 'Implements';
+    }
+    return 'Types';
+  };
+  const getTypes = () => {
```
Comment (marckong) [inline packages/insomnia/src/ui/components/graph-ql-explorer/graph-ql-explorer-type.tsx:29]: nit pick here - same with the above comment in regards to locally declared function that does not mutate any of the React state inside.

File: packages/insomnia/src/ui/components/graph-ql-explorer/graph-ql-explorer-type.tsx (hunk 2 above)
Comment (jackkav) [reply packages/insomnia/src/ui/components/graph-ql-explorer/graph-ql-explorer-type.tsx:17]: Its the closest thing to the original implementation that separates the 3 concerns of title, type and fallback to null.

File: packages/insomnia/src/ui/components/graph-ql-explorer/graph-ql-explorer-type.tsx [hunk 4]
```diff
@@ -15,94 +13,64 @@ interface Props {
...
+          <ul className="graphql-explorer__defs">
+            {types.map(type => (
+              <li key={type.name}>
+                <GraphQLExplorerTypeLink onNavigate={onNavigateType} type={type} />
+              </li>
+            ))}
+          </ul>
+        </Fragment>
+        : null}
+      {sortedFields.length
```
Comment (marckong) [inline packages/insomnia/src/ui/components/graph-ql-explorer/graph-ql-explorer-type.tsx:64]: It seems GraphQLExplorerFieldsList already handles empty array case. The previous implementation would show the Fields section despite nothing is there. This implementation will not show anything. Would this be the intention?

Current:
image
image

Before:
image

Comment (jackkav) [reply packages/insomnia/src/ui/components/graph-ql-explorer/graph-ql-explorer-type.tsx:64]: I can't get this reproduction, but it did do this before my second pass commit, can you double check your branch?

Comment (filfreire) [reply packages/insomnia/src/ui/components/graph-ql-explorer/graph-ql-explorer-type.tsx:64]: @jackkav @marckong I also can't reproduce this with latest commit:
image

**Review by filfreire — APPROVED** [review]
LGTM! 👍

**Pull request merged**